storage/namespace.go

```diff
@@ -184,7 +203,11 @@ func newDatabaseNamespace(
 
 	n.initShards(nopts.NeedsBootstrap())
 
-	return n
+	return n, nil
+}
+
+func (n *dbNamespace) Options() namespace.Options {
```

Comment:
nit: might as well just cache the metadata options to save function call overhead in many places.